ngraph/test/type_prop/experimental_detectron_roi_feature_extractor.cpp

```diff
@@ -169,3 +169,192 @@ TEST(type_prop, detectron_roi_feature_extractor_intervals)
         ASSERT_TRUE(roi->get_output_partial_shape(0).same_scheme(ref_out_shape));
     }
 }
+
+struct ROIFeatureIntervalsTestParams
+{
+    PartialShape input_shape;
+    Dimension channels[4];
+    Dimension first_dims[4];
+};
+
+void run_roi_feature_test(const ROIFeatureIntervalsTestParams& params)
+{
+    Attrs attrs;
+    attrs.aligned = false;
+    attrs.output_size = 14;
+    attrs.sampling_ratio = 2;
+    attrs.pyramid_scales = {4, 8, 16, 32};
+
+    auto layer0_channels = params.channels[0];
+    auto layer1_channels = params.channels[1];
+    auto layer2_channels = params.channels[2];
+    auto layer3_channels = params.channels[3];
+
+    auto layer0_shape = PartialShape{params.first_dims[0], layer0_channels, 200, 336};
+    auto layer1_shape = PartialShape{params.first_dims[1], layer1_channels, 100, 168};
+    auto layer2_shape = PartialShape{params.first_dims[2], layer2_channels, 50, 84};
+    auto layer3_shape = PartialShape{params.first_dims[3], layer3_channels, 25, 42};
+
+    auto expected_channels = layer0_channels & layer1_channels & layer2_channels & layer3_channels;
+
+    auto ref_out_shape = PartialShape{params.input_shape[0], expected_channels, 14, 14};
+
+    auto input = std::make_shared<op::Parameter>(element::f32, params.input_shape);
+    auto pyramid_layer0 = std::make_shared<op::Parameter>(element::f32, layer0_shape);
+    auto pyramid_layer1 = std::make_shared<op::Parameter>(element::f32, layer1_shape);
+    auto pyramid_layer2 = std::make_shared<op::Parameter>(element::f32, layer2_shape);
+    auto pyramid_layer3 = std::make_shared<op::Parameter>(element::f32, layer3_shape);
+
+    auto roi = std::make_shared<ExperimentalROI>(
+        NodeVector{input, pyramid_layer0, pyramid_layer1, pyramid_layer2, pyramid_layer3}, attrs);
+
+    ASSERT_EQ(roi->get_output_element_type(0), element::f32);
+    ASSERT_TRUE(roi->get_output_partial_shape(0).same_scheme(ref_out_shape));
+}
+
+struct ROIFeatureIntervalsTest : ::testing::TestWithParam<ROIFeatureIntervalsTestParams>
+{
+};
+
+TEST_P(ROIFeatureIntervalsTest, detectron_roi_feature_extractor_intervals_1)
+{
+    run_roi_feature_test(GetParam());
```

Comment:
I don't understand why do you need to have an additional method `run_roi_feature_test`? Looks like it is a redundant function.
You can have all code in the test body.

Reply:
Done.